packages/xo-server/src/xo-mixins/subjects.js

```diff
@@ -231,6 +248,74 @@ export default class {
     })
   }
 
+  // New implementation of registerUser that:
+  //   - allows multiple providers per XO user
+  //   - binds a XO user to the provider's user with a unique ID
+  // - id: the ID that the provider uses to identify the user
+  // - name: the name of the user according to the provider
+  // - data: additional data about the user that the provider may want to store
+  async registerUser2(providerId, { user: { id, name }, data }) {
+    const users = await this.getAllUsers()
+
+    // Get the XO user bound to the provider's user
+    let user = users.find(user => user.authProviders?.[providerId]?.id === id)
+
+    // If that XO user doesn't exist or doesn't have the correct username, there
+    // is a chance that there is another XO user that already has that username
+    let conflictingUser
+    if (user?.email !== name) {
+      conflictingUser = users.find(user => user.email === name)
+
+      if (conflictingUser !== undefined) {
+        if (!this._xo._config.authentication.mergeProvidersUsers) {
```

Comment:
Please, document this new settings similarly than: https://github.com/vatesfr/xen-orchestra/blob/ccd34c1610f9eabae3c28adc9b6a552ec45c7e5e/packages/xo-server/config.toml#L45-L46